packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp

```diff
@@ -4141,6 +4150,16 @@ namespace Tpetra {
           rowValues[j] *= scaleValue;
         }
       }
+
+      // FENCE REVIEW - NOT TESTED
+      //   Testing: This code is NOT exercised by unit tests.
+      //   GTX960:  N/A
+      //   White:   N/A
+      //   Plan:    Potentially add unit testing for this code.
+      //            Potentially remove this fence but prior code appears
+      //            to need refactoring.
+      //   Notes:
+
```

Comment:
@kddevin This can be a design choice. We need to decide a API principle. In this code, we put fence in entering the code and also put the fence exiting the code. We should choose either of them. If we can make sure all Tpetra code assume that the code is non-blocking, we can leave the exiting as open-end. If we assume that all Tpetra code is blocking and synchronize when exiting, we do not need to put fence at the beginning of the API. 

We need to apply some consistency rules. According to the design principle, we can remove either of them.

Reply:
Removed with PR 7560